### ISSUE TITLE
Upgrade zammad 6.1 to 6.2 missing external datasource field type and issue #4825 still there

### ISSUE THREAD

**Issue by tgb-sw** [zammad/zammad#4983]
### Used Zammad Version

6.2

### Environment

- Installation method: package
- debian 11
- Database + version:  mariadb 10.5]
- Elasticsearch version: 7.10.2
- Browser + version: firefox 120

### Actual behaviour

We upgraded with apt upgrade.
After the upgrade we encounter that the core workflow action "replace text when empty" still replaces text even if not empty (see issue #4825 
Also the new field format "external data source" is missing when new attributes arge created.

### Expected behaviour

#4825 should work and the new field format should be available

### Steps to reproduce the behaviour

We have a snapshot of zammad before the upgrade.
We tested the upgrade multiple times. No errors are displayed, but the functionality is missing.
zammad (6.2.0-1702303938.81e0c1fd.bullseye) wird eingerichtet ...
# Enforcing Redis...
# Creating Redis bootstart
Synchronizing state of redis-server.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable redis-server
# Starting Redis server
# Redis server is running.
# (Re)creating init scripts
Nothing to do.
Nothing to do.
Nothing to do.
# Enabling Zammad on boot
# Stopping Zammad
# Clear cache...
# database.yml found. Updating db...
== 20220920084800 SettingAddStoreProviderS3: migrating ========================
== 20220920084800 SettingAddStoreProviderS3: migrated (0.0136s) ===============

== 20230829085650 LdapSourceConfig: migrating =================================
== 20230829085650 LdapSourceConfig: migrated (0.0215s) ========================

== 20230830100819 CreateSSLCertificates: migrating ============================
-- create_table(:ssl_certificates)
   -> 0.0067s
== 20230830100819 CreateSSLCertificates: migrated (0.0078s) ===================

== 20230904123822 SetIdoitSSLDefault: migrating ===============================
== 20230904123822 SetIdoitSSLDefault: migrated (0.0009s) ======================

== 20230905073448 SetGitLabSSLDefault: migrating ==============================
== 20230905073448 SetGitLabSSLDefault: migrated (0.0074s) =====================

== 20230905074116 SetElasticSearchSSL: migrating ==============================
== 20230905074116 SetElasticSearchSSL: migrated (0.0104s) =====================

== 20230906162437 SetMailSSLDefault: migrating ================================
== 20230906162437 SetMailSSLDefault: migrated (0.0400s) =======================

== 20230912080534 GroupHierarchy: migrating ===================================
-- change_column(:groups, :name, :string, {:limit=>970})
   -> 0.0049s
-- add_column(:groups, :name_last, :string, {:limit=>160, :null=>true})
   -> 0.0043s
-- add_column(:groups, :parent_id, :integer, {:null=>true})
   -> 0.0043s
-- add_foreign_key(:groups, :groups, {:column=>:parent_id})
   -> 0.1606s
-- change_column(:groups, :name_last, :string, {:limit=>160, :null=>false})
   -> 0.0192s
== 20230912080534 GroupHierarchy: migrated (1.1180s) ==========================

== 20231011055535 SamlSignEncrypt: migrating ==================================
== 20231011055535 SamlSignEncrypt: migrated (0.0219s) =========================

== 20231016120344 SearchIndexSettings: migrating ==============================
== 20231016120344 SearchIndexSettings: migrated (0.0059s) =====================

== 20231017091313 LoopProtectionSettings: migrating ===========================
== 20231017091313 LoopProtectionSettings: migrated (0.0181s) ==================

== 20231024103209 Issue4838DataRetention: migrating ===========================
-- add_column(:jobs, :object, :string, {:limit=>100, :null=>true})
   -> 0.0031s
-- change_column(:jobs, :object, :string, {:limit=>100, :null=>false})
   -> 0.0142s
== 20231024103209 Issue4838DataRetention: migrated (0.0221s) ==================

== 20231107090334 SimpleStorageConfigurationCheck: migrating ==================
== 20231107090334 SimpleStorageConfigurationCheck: migrated (0.0076s) =========
# Updating translations...
# No custom packages detected...
# Configuring Elasticsearch...
-> removing [ingest-attachment]...
-> Installing ingest-attachment
-> Downloading ingest-attachment from elastic
^M[                                                 ] 1%   ^M[>                                                ] 2%   ^>@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@     WARNING: plugin requires additional permissions     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
* java.lang.RuntimePermission accessClassInPackage.sun.java2d.cmm.kcms
* java.lang.RuntimePermission accessDeclaredMembers
* java.lang.RuntimePermission getClassLoader
* java.lang.reflect.ReflectPermission suppressAccessChecks
* java.security.SecurityPermission createAccessControlContext
* java.security.SecurityPermission insertProvider
* java.security.SecurityPermission putProviderProperty.BC
See http://docs.oracle.com/javase/8/docs/technotes/guides/security/permissions.html
for descriptions of what these permissions allow and the associated risks.

-> Installed ingest-attachment
# Enforcing 0600 on database.yml ...
# Starting Zammad
Module proxy already enabled
Considering dependency proxy for proxy_http:
Module proxy already enabled
Module proxy_http already enabled
Considering dependency proxy for proxy_wstunnel:
Module proxy already enabled
Module proxy_wstunnel already enabled
# Creating webserver bootstart
Synchronizing state of apache2.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable apache2
# Restarting webserver apache2
####################################################################################

Add your fully qualified domain name or public IP to servername directive of
apache2, if this installation is done on a remote server. You have to change:
/etc/apache2/sites-available/zammad.conf and restart apache2 process.
Otherwise just open http://localhost/ in your browser to start using Zammad.

####################################################################################

We are unable to repdroduce this in a fresh installation.

### I'm sure this is a bug and no feature request or a general question.

no

**Comment by MrGeneration** [comment]
Kindly read the release notes.
Thank you.

https://zammad.com/en/releases/6-2

**Comment by tgb-sw** [comment]
Thank you very much!

I did read the release notes up and down multiple times....
The only thing I can imagine was about the database. 
Maybe my english is to bad, but the release notes only mention "but must be migrated to PostgreSQL until the release of Zammad 7." There is no note about missing features in Version 6.2 ....

**So I migrated from mariadb to postgres. And without any further changes, all problems are solved!**
